app/services/full_refresh_upload_task_service.rb

```diff
@@ -8,7 +8,13 @@ def process
     end
 
     @task = FullRefreshUploadTask.create!(task_options)
-    Source.update(source_id, :refresh_started_at => Time.current)
+    Source.update(source_id,
+                  :refresh_started_at  => Time.current,
+                  :refresh_finished_at => nil,
```

Comment:
@hsong-rh If the Upload task ends in an error we would have to set the refresh_finished_at and set the refresh_state to error